keyboards/nckiibs/vaguettelite/vaguettelite.h

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2020 noclew
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT_vaguette_lite( \
+    k01, k02, k03, k04, k05, k06, k07, k08, k09, k10, k11, k12, k65, k66, k69,\
+    k13, k14, k15, k16, k17, k18, k19, k20, k21, k22, k23, k24, k67, k68,\
+    k25, k26, k27, k28, k29, k30, k31, k32, k33, k34, k35, k36, k61, k70,\
+    k60, k37, k38, k39, k40, k41, k42, k43, k44, k45, k46, k47, k48, k62,\
+    k49, k50, k51, k52, k53, k54, k55, k56, k57, k58, k59, k63, k64 \
```

Comment:
Trying to wrap my head around this.

Let me guess: `k54` is used and `k52`, `k53`, `k55` and `k56` are unused if you use a full-length (6.25u) spacebar; then vice versa if you use a split spacebar?

Reply:
Yes. The board supports multiple layouts including 6.25u, 2 x 2u, 1 x 12u
spacebar :)

Reply:
K, `LAYOUT_vaguette_lite` should be replaced with `LAYOUT_all` everywhere inside `keyboards/nckiibs/vaguettelite/`, for starters. I'll give a full review a bit later.

Reply:
> K, `LAYOUT_vaguette_lite` should be replaced with `LAYOUT_all` everywhere inside `keyboards/nckiibs/vaguettelite/`, for starters. I'll give a full review a bit later.

Changed accordingly. By the way, can you please let me know why it has to be LAYOUT_all?

Reply:
> By the way, can you please let me know why it has to be LAYOUT_all?

Strictly speaking, it doesn't *have to* be in a technical/functional sense, but it's the preferred naming scheme for when a keyboard supports more than one physical layout.

[QMK Keyboard Guidelines - QMK Docs](https://docs.qmk.fm/#/hardware_keyboard_guidelines?id=ltkeyboard_namehgt)

I don't typically enforce the additional keymaps being present in the pull request, but I very much prefer both (1) that those guidelines be followed for the names of the layout macros, because it is quite complicated to change them after the fact (users of the board will already be used to a name that may be "wrong"), and (2) that the layout macro names are descriptive of what the layout is.

Knowing that this macro supports every location where it is possible to have a switch is helpful for us if a user makes contact and has built their keyboard with a different layout than is defined – we can add the layout after the fact without making disruptive changes to the code that is already there.

Reply:
Ah, I see. Much appreciated for the detailed explanation. I will make an additional keymap and update you.

Reply:
If you're willing, please do so.